.github/workflows/continuous-monitoring.yml

```diff
@@ -0,0 +1,90 @@
+name: Continuous Monitoring of Dockerhub and public ECR
+on:
+  schedule:
+    - cron:  '0 */6 * * *'
+
+jobs:
+  monitor-ecr:
+    name: Monitor public ECR
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Login to public ECR
+        uses: docker/login-action@v1
+        with:
+          registry: public.ecr.aws
+
+      - name: Pull an image from public ECR
+        run: docker pull public.ecr.aws/xray/aws-xray-daemon:latest
+
+      - name: Publish metric on pulling an image
+        if: ${{ always() }}
+        run: |
+          if [[ "$(docker images -q public.ecr.aws/xray/aws-xray-daemon:latest 2> /dev/null)" == "" ]]; then
+            aws cloudwatch put-metric-data --metric-name PullImageFailureFromECR --dimensions failure=rate --namespace MonitorDaemon --value 1 --timestamp $(date +%s)
+          else
+            aws cloudwatch put-metric-data --metric-name PullImageFailureFromECR --dimensions failure=rate --namespace MonitorDaemon --value 0 --timestamp $(date +%s)
+          fi
+
+      - name: Run daemon image
+        run: |
+          docker run --name xray-daemon public.ecr.aws/xray/aws-xray-daemon:latest -n us-west-2 &
+          sleep 30
+
+      - name: Publish metrics on daemon startup
+        if: ${{ always() }}
+        run: |
+          if [[ "$(docker container inspect -f '{{.State.Status}}' xray-daemon )" != "running" ]]; then
+            aws cloudwatch put-metric-data --metric-name DaemonImageStartupFailureFromECR --dimensions failure=rate --namespace MonitorDaemon --value 1 --timestamp $(date +%s)
+          else
+            aws cloudwatch put-metric-data --metric-name DaemonImageStartupFailureFromECR --dimensions failure=rate --namespace MonitorDaemon --value 0 --timestamp $(date +%s)
+          fi
+
+  monitor-dockerhub:
+    name: Monitor docker hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Login to DockerHub
```

Comment:
Should we verify without login?

Note, for public ECR I guess most pulls might be authenticated so maybe not as big of a deal, for docker hub I suspect many pulls are unauthenticated. But probably good to pull unauthenticated for both, I guess once every 6 hours should fall in anonymous quota